### ISSUE TITLE
[Feature] Iris Learning room usage

### ISSUE THREAD

**Issue by CommanderStorm** [TUM-Dev/NavigaTUM#364]
**Is your feature request related to a problem? Please describe.**
Currently, we don't do a good job communicating learning-room availability to our users.

**Describe the solution you'd like**
If on the StudiTUM page, it would be really nice, if a separate card with information like this could show up:
![Screenshot_20230124_130229_Campus](https://user-images.githubusercontent.com/26258709/214286772-71c5f58c-308e-428b-9b82-13ff116bb3ba.jpg)
(Screenshot of the android Campus App)

**Describe alternatives you've considered**
Just linking to the usage, like we currently do

**Additional context**
This is not a high-priority feature and should be based on #200

**Comment by CommanderStorm** [comment]
> *This was generated by AI during triage.*

## Agent Brief

**Category:** enhancement
**Summary:** Show a learning-room availability card on building / area pages by integrating the AStA Iris API, gated by a build-time coverage flag and disclosed in the data-protection page.

**Current behavior:**
NavigaTUM does not surface learning-room availability. Users who want to know whether a StudiTUM (or other) learning room is currently free have to leave NavigaTUM and consult the AStA Iris site or the TUM Campus App.

**Desired behavior:**
On every building / area page whose ID is known (at data-build time) to have Iris coverage, render a card listing that building's learning rooms with their live availability. For each room:

- Show the room name, status, and (where applicable) `belegung_durch` / `belegung_bis` (booked by / booked until).
- For `WAAS` (Wi-Fi-counter) rooms, show the occupancy percentage and colour band that Iris provides.
- Link each row to the corresponding NavigaTUM room detail page, resolved via alias lookup (see below).
- The card refreshes every 60 seconds while it is visible to the user; polling pauses when the card is off-screen or the tab is hidden.
- Status labels render in English on EN pages (`frei`→"free", `belegt`→"booked", `unbekannt`→"unknown", `WAAS`→"sensor-based"); German is passed through.
- If the Iris API is unreachable or returns a malformed response, the card degrades silently — it does not block the rest of the page from rendering.

The webclient **must not** hit the Iris API on pages whose building ID is not in the coverage set. That gate is the only thing standing between us and 200+ wasted fetches every time someone lands on an arbitrary room page.

**Data-side integration (new):**

The data pipeline gains a step that, on each build, fetches the Iris response once and records the set of NavigaTUM building IDs that have at least one room present (joined via `raum_nr_architekt` → `aliases`). This set is exposed through the existing entry data (e.g. as a boolean `has_iris_coverage` on each building / area entry, or — if multiple per-page integrations follow the same pattern — as a more general `external_data_sources` flag map). The webclient gates the Iris-availability card on this flag.

The coverage set refreshes at data-build cadence (Iris's room roster is stable on the order of months — minute-to-minute changes are the live `status` field, which always comes from the runtime API). If Iris is unreachable at data-build time, the build does not fail; it logs a warning and falls back to the previously-known coverage set (or to an empty set on first build), so a transient AStA outage cannot break our deployment.

**Data-protection integration (new):**

Because the live status is fetched browser-side, the user's IP and request are exposed to `iris.asta.tum.de` (AStA TUM). This must be disclosed in NavigaTUM's data-protection page before the feature ships:

- Name the endpoint (`iris.asta.tum.de`) and the operator (AStA der TUM).
- State what is sent (the request itself, the user's IP, standard browser headers) and what is **not** sent (no NavigaTUM identifiers, no query the user typed elsewhere on the site).
- State the purpose (showing live learning-room availability on building pages that have it).
- Link the AStA privacy page if one exists; if not, this is a flag for the human reviewer rather than a blocker for the agent.
- The disclosure should be added in a way that scales — future browser-side third-party integrations (e.g. Transitous, already in use; potentially others) should slot into the same section rather than each getting its own ad-hoc paragraph.

**Key interfaces:**

- **Iris API:** `GET https://iris.asta.tum.de/api/` — public, no auth, `Access-Control-Allow-Origin: *` so it is browser-callable directly. Same pattern as the Transitous `/stoptimes` endpoint already in use. **Do not** proxy live status through the NavigaTUM server.
- **Response shape:** top-level `{raeume: [...], gruppen: [...], bibdata: ...}`. Each `raeume[]` entry exposes:
  - `raum_nr_architekt` — the join key into NavigaTUM (e.g. `DG.29@3515`, `MW 1450@5504`). Matches the raw `<arch_name>@<building_id>` form already present in each NavigaTUM room's `aliases` list.
  - `gebaeude_code` — matches NavigaTUM building IDs 1:1 (verified across all 27 buildings currently in Iris).
  - `status` — one of `frei`, `belegt`, `WAAS`, `unbekannt`.
  - `belegung_ab` / `belegung_bis` / `belegung_durch` / `belegung_fuer` / `belegung_in` — booking window / duration / booker; often empty strings.
  - `attribute` — features like `Whiteboard`, `Sonnenblende`.
  - WAAS rooms additionally carry `color`, `percent`, `subtitle`.
- **Data-pipeline coverage step:** a new processor under `/data/processors/` (or equivalent) that fetches Iris once per build, derives the set of NavigaTUM building IDs with at least one matched room, and persists that into the entry data. The exact field shape (per-entry boolean vs. central manifest vs. `external_data_sources` map) is the agent's call; the constraint is that the webclient can decide per page, without an extra round trip, whether to render the card.
- **Room → detail-page link:** look each Iris room up by `raum_nr_architekt` in NavigaTUM's `aliases`. If no match, omit the row (currently exactly one room — the Stammgelände Teilbibliothek `1140@0501` — has no match, and that is acceptable).
- **Card placement:** the existing building / area page-section system, alongside cards like the recently added nearby-public-transport one. Match those visual conventions (note: details-page section cards use `rounded-sm`, not the `rounded-lg` of the Motis family).
- **Data-protection page:** the existing privacy / data-protection page in the webclient. New third-party-endpoint disclosure follows whatever structure that page already uses.

**Acceptance criteria:**

- [ ] Data pipeline produces a stable per-entry signal (boolean flag or equivalent) indicating Iris coverage for each building / area entry.
- [ ] A data-build run with the Iris API reachable populates the signal for the buildings currently in Iris's response (27 at time of writing); a build with the Iris API unreachable does not fail and falls back to the prior coverage set (or an empty set on first build).
- [ ] On building / area pages where the signal is true, the card renders with one row per Iris room for that building (excluding rooms that fail the alias lookup).
- [x] On building / area pages where the signal is false, **no request to `iris.asta.tum.de` is issued** (verified by browser-side test or by inspecting network calls in a component test).
- [x] Each row shows room name, status, and (for `belegt` rooms) the booker / booked-until information.
- [x] WAAS rooms show the occupancy percentage and colour from the Iris response.
- [ ] Rows are clickable links to the corresponding NavigaTUM room detail page.
- [x] The card polls Iris every 60 seconds while visible; polling pauses when off-screen or the tab is hidden.
- [x] Status labels render in English on EN pages, German on DE pages.
- [ ] If the runtime Iris fetch fails (network error, non-2xx, malformed response), the card hides itself or shows a small inline error and does not break the rest of the page.
- [ ] The data-protection page is updated to disclose `iris.asta.tum.de` as a third-party endpoint, what is sent, and the purpose; the disclosure is structured so further browser-side third-party integrations can be added next to it without duplication.
- [ ] Component test coverage exists for the parsing + alias-lookup logic, using a small fixture from the live Iris response.

**Out of scope:**

- Historical occupancy data, predictions, or trend charts — only the current snapshot.
- Backfilling NavigaTUM data to cover rooms Iris reports but NavigaTUM does not (e.g. the Stammgelände Teilbibliothek). Treat those as silent misses.
- Server-side proxying or caching of **live** Iris status — the runtime call stays browser-side. Build-time coverage discovery is server-side; live polling is not.
- A real-time / push coverage feed — coverage updates at the existing data-build cadence.
- Any contribution back to the Iris dataset itself.
- Surfacing the `attribute` list (whiteboard / sun blinds / …) on the card — availability only.
- Surfacing the top-level `gruppen` or `bibdata` fields from the Iris response.
- Rewriting the data-protection page beyond adding the Iris disclosure (and making the section extensible for future similar integrations).
